lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java

```diff
@@ -39,13 +39,24 @@
 
   IFusedLocationProviderService service;
 
-  private IFusedLocationProviderCallback.Stub remoteCallback
+  IFusedLocationProviderCallback.Stub remoteCallback
       = new IFusedLocationProviderCallback.Stub() {
     public void onLocationChanged(final Location location) throws RemoteException {
+
+      if (service == null) {
+        throw new IllegalStateException("Location update received after client was "
```

Comment:
I know its not your code but can you also remove this exception? Its too aggressive when compared to Play Services and silently ignoring the location update as you've done below is a fine solution.